web/src/stream_edit.ts

```diff
@@ -747,6 +766,39 @@ export function initialize(): void {
         $(".dialog_submit_button").attr("data-stream-id", stream_id);
     });
 
+    $("#channels_overlay_container").on("click", ".reactivate", function (this: HTMLElement, e) {
+        e.preventDefault();
+        e.stopPropagation();
+
+        function do_unarchive_stream(): void {
+            const stream_id = Number.parseInt(
+                $(".dialog_submit_button").attr("data-stream-id")!,
+                10,
+            );
+            unarchive_stream(stream_id, $(".stream_change_property_info"));
+        }
+
+        const stream_id = get_stream_id(this);
```

Comment:
`stream_id` is being computed in two places - once here and then again in `do_unarchive_stream`. It is fine to just compute it once.